kernel/src/ipc.rs

```diff
@@ -27,22 +34,8 @@ pub enum IPCUpcallType {
 }
 
 /// State that is stored in each process's grant region to support IPC.
-struct IPCData<const NUM_PROCS: usize> {
-    /// An array of process buffers that this application has shared
-    /// with other applications.
-    shared_memory: [ReadWriteProcessBuffer; NUM_PROCS],
-    search_buf: ReadOnlyProcessBuffer,
-}
-
-impl<const NUM_PROCS: usize> Default for IPCData<NUM_PROCS> {
-    fn default() -> IPCData<NUM_PROCS> {
-        const DEFAULT_RW_PROC_BUF: ReadWriteProcessBuffer = ReadWriteProcessBuffer::const_default();
-        IPCData {
-            shared_memory: [DEFAULT_RW_PROC_BUF; NUM_PROCS],
-            search_buf: ReadOnlyProcessBuffer::default(),
-        }
-    }
-}
+#[derive(Default)]
+struct IPCData;
```

Comment:
So, I left the empty struct since the capsule are most likely going to get monmorphized anyway since they most likely won't share the same number of subscribe, ro allow, and rw allow counts. I think it would be pretty rare to share. At least that is the approach I used for the capsule conversion process.

I could be convinced that IPC was special and to store `()` as the grant type though.